protocols/imap/src/main/java/org/apache/james/imap/processor/AuthenticateProcessor.java

```diff
@@ -155,4 +165,16 @@ protected Closeable addContextToMDC(AuthenticateRequest request) {
             .addToContext("authType", request.getAuthType())
             .build();
     }
+
+    private void doSASL(String jwtToken, ImapSession session, ImapRequest request, Responder responder) {
+        Optional<String> kid = JwtTokenHelper.getHeaderValueWithoutVerifySignature(jwtToken, "kid", String.class);
+        if (kid.isEmpty()) {
+            // fail
```

Comment:
Use a `null` kid instead of failing.

Reply:
Once again what if for some reason the OIDC is referencing multiple keys?

Reply:
Key rotation for instance: you might keep the older key for a few days to avoid invalidating all of the tokens signed with it.

Reply:
thus if the kid is missing it should be failing IMO

Reply:
No not really you just care about one matching cf https://github.com/apache/james-project/pull/792/commits/ded4c3056fab9ff700b70871c79a77ac63db28ba

Reply:
So you try to verify against each key then until you have a match correct? Then alright yes